### PR TITLE
add tree-sitter-indent to melpa

### PR DESCRIPTION
**Pull request by FelipeLema**
### Brief summary of what the package does

Use Tree-sitter as backend to source code indentation.

### Direct link to the package repository

https://codeberg.org/FelipeLema/tree-sitter-indent.el

### Your association with the package

Author and maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

CC'ing @theothornhill because "they made me do this"

### CODE REVIEW

File: recipes/tree-sitter-indent [hunk 1]
```diff
@@ -0,0 +1,3 @@
+(tree-sitter-indent :fetcher git
+                    :url "https://codeberg.org/FelipeLema/tree-sitter-indent.el.git"
+                    :branch "main")
```
Comment (riscy) [inline recipes/tree-sitter-indent:3]: Just quickly noting - if you're specifying `:branch "main"` for compatibility with straight.el, and `main` is your default branch, I believe straight.el has fixed this.  Generally we discourage specifying branch except in unusual cases.

**Comment by alphapapa** [comment]
@FelipeLema FYI, I see that you're using `emacs-sandbox.sh`: https://codeberg.org/FelipeLema/tree-sitter-indent.el/commit/af0abb082d5a751d09058d2a057b8737ae76646f  You may like to use `makem.sh` instead: https://github.com/alphapapa/makem.sh  It includes the same kind of sandbox functionality but is especially designed for testing and linting packages.  You could replace the bespoke code in your `Makefile` that runs `package-lint` with, e.g. `make sandbox=t install-linters=t lint-package`, or `./makem.sh --sandbox --install package-lint lint-package`.

File: recipes/tree-sitter-indent (hunk 1 above)
Comment (FelipeLema) [reply recipes/tree-sitter-indent:3]: oh, I see... I'll remove this property, then.

**Comment by FelipeLema** [comment]
@alphapapa wellwellwell... I didn't know about `makem.sh` and indeed it looks like it fulfills my needs.

Using it would be blocking for this PR? It would take me a while to change my scripts to it.

**Comment by riscy** [comment]
> Using it would be blocking for this PR?

That's not a blocker, but I'd recommend it too.  In any case this looks interesting.  I think there are no blockers, but some small suggestions I'd make:
- tree-sitter-indent.el#L466: It's safer to sharp-quote function names; use `#'`
- tree-sitter-indent.el#L282: Optionally use `unless ...` instead of `when (not ...)`
- tree-sitter-indent.el#L275: This double negation can be collapsed (`not` aliases `null`)
- tree-sitter-indent.el#L438: This double negation can be collapsed (`not` aliases `null`)

**Pull request merged**